plugins/linear-genome-view/src/LinearGenomeView/components/SequenceDialog.tsx

```diff
@@ -46,21 +46,36 @@ const useStyles = makeStyles(theme => ({
 
 /**
  * Fetches and returns a list features for a given list of regions
- * @param selectedRegions - Region[]
- * @returns Features[]
  */
 async function fetchSequence(
-  self: LinearGenomeViewModel,
+  model: LinearGenomeViewModel,
   selectedRegions: Region[],
+  opts: {
+    signal?: AbortSignal
+  },
 ) {
-  const session = getSession(self)
-  const assemblyName =
-    self.leftOffset?.assemblyName || self.rightOffset?.assemblyName || ''
-  const { rpcManager, assemblyManager } = session
-  const assemblyConfig = assemblyManager.get(assemblyName)?.configuration
+  const session = getSession(model)
+  const { leftOffset, rightOffset } = model
 
-  // assembly configuration
-  const adapterConfig = readConfObject(assemblyConfig, ['sequence', 'adapter'])
+  if (!leftOffset || !rightOffset) {
+    throw new Error('no offsets on model to use for range')
+  }
+
+  if (leftOffset.assemblyName !== rightOffset.assemblyName) {
+    throw new Error(
+      'not able to fetch from multiple assemblies at once currently',
```

Comment:
I feel like this error message might be confusing to a user. Specifically "currently" could make it seem like it failed because a resource was unavailable. Maybe something like this?

```suggestion
      'not able to fetch sequences from multiple assemblies',
```

Reply:
:+1: sometimes i do a weird thing where i try to imply that "we could add this functionality" as a hint to the user but it's probably just better to give a solid error